lib/puppet/functions/shell_escape.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'shellwords'
```

Comment:
Not sure if this should be moved into the `shell_escape` method.  With newer versions of Ruby it might otherwise trigger https://github.com/puppetlabs/puppet-strings/issues/296

Reply:
Ah, I was hesitant to do this in order to avoid including code at load time but only when used.  Another good reason to do so, i will update it!